crates/nargo_cli/src/cli/mod.rs

```diff
@@ -55,13 +56,14 @@ enum NargoCommand {
     Verify(verify_cmd::VerifyCommand),
     Test(test_cmd::TestCommand),
     Gates(gates_cmd::GatesCommand),
+    Lsp(lsp_cmd::LspCommand),
 }
 
 pub fn start_cli() -> eyre::Result<()> {
     let NargoCli { command, mut config } = NargoCli::parse();
 
     // Search through parent directories to find package root if necessary.
-    if !matches!(command, NargoCommand::New(_)) {
+    if !matches!(command, NargoCommand::New(_)) && !matches!(command, NargoCommand::Lsp(_)) {
```

Comment:
```suggestion
    if !matches!(command, NargoCommand::New(_) | NargoCommand::Lsp(_)) {
```

This is equivalent, no?

Reply:
Probably, I didn't think too much about this 🙈